src/main/java/de/muenchen/isi/domain/service/InfrastruktureinrichtungService.java

```diff
@@ -112,6 +114,30 @@ public void deleteInfrastruktureinrichtungById(final UUID id)
         this.infrastruktureinrichtungRepository.deleteById(id);
     }
 
+    /**
+     * Die Methode gibt {@link InfrastruktureinrichtungListElementModel}e sortiert nach {@link InfrastruktureinrichtungTyp} und Name der Einrichtung zurück.
+     *
+     * @param bauvorhabenId um Bauvorhaben zu ermitteln
+     * @return die {@link InfrastruktureinrichtungListElementModel}s sortiert in aufsteigender Reihenfolge.
+     */
+    public InfrastruktureinrichtungListElementsModel getAllReferencedInfrastruktureinrichtungForBauvorhaben(
+        final UUID bauvorhabenId
+    ) {
+        final var infrastruktureinrichtungListElementsModel = new InfrastruktureinrichtungListElementsModel();
+        final List<InfrastruktureinrichtungListElementModel> listElements =
+            this.infrastruktureinrichtungRepository.findAllByBauvorhabenId(bauvorhabenId)
```

Comment:
Die Infrastuktureinrichtungen können bereits bei der Extraktion aus der Datenbank soritert werden.

Falls möglich bereits durch das SQL-Statement in der Datenbank sortieren lassen, da diese Sortierung in der Datenbank hochperformant ist und code in der Anwendung spart.